### DISCUSSION TITLE
WebDAV unreliable/unstable (on Windows)

### DISCUSSION THREAD

**Discussion by Stitch10925** [seaweedfs/seaweedfs#2898] [Q&A]
I've been testing out SeaweedFS on and off for the last week or so. My goal is to be able to have easily expandable, redundant storage for Linux and Windows.

FUSE Mount on Linux works like a dream. Really fast and reliable for small and large files. Even streaming of movies works without a hitch.

For Windows the story is different. There is no FUSE Mount for Windows. There are ports, though. The main one I found was WinFSP. I managed to connect to the master using it, but was only able to access the weedFS root file system. I could not figure out how to access the files that I had uploaded.

This leaves WebDAV. WebDAV, however, was very unreliable. When trying to access large files or would suddenly give an error saying the file did not exist. When uploading a bunch of Pdf's and then trying to open them, the Pdf's would just be blank. Uploading large files would initially go really fast, then would hang at the end and never complete. Basically WebDAV is useless. (This was on a Win10 machine).

Currently I am sharing a FUSE Mount via Samba from a LXC. This seams to work quite well, however, I am new to Linux and setting permissions and accounts to access the Samba share is a pain to me. It also bothers me that this adds an extra hop to access the files.

I was wondering if anyone has had similar, better or worse experiences with WebDAV or accessing WeedFS from Windows, or if anyone might have suggestions or ideas on other ways to tackle the Windows access issues I'm having?

**Comment by chrislusf** [comment]
Haven't started to work with WinFSP yet.

WebDAV is based on go standard webDAV package and everything is just in one file https://github.com/chrislusf/seaweedfs/blob/master/weed/server/webdav_server.go
It hasn't been on the priority list yet.

**Reply by Stitch10925** [reply]
I installed WinFSP and WinFSP/SSHFS-Win, I then used it to connect to the master using:
`net use Z: \\sshfs\root@<master ip>`

When I do this it connects but gives me the following folder structure:

![image](https://user-images.githubusercontent.com/69055295/162689027-17e9e274-52fd-4772-af9a-cd0b0930a5d6.png)

I haven't been able to figure out how to get to the file I actually uploaded.

**Comment by Stitch10925** [comment]
Issue might have been resolved in this ticket:
https://github.com/chrislusf/seaweedfs/issues/3217

Haven't tested it yet, however.
